src/js/main.js

```diff
@@ -1,77 +1,114 @@
+    if ($search.value.length <= 2 && $searchType.value != 5) {
+      alert('Please enter at least 3 characters');
+      e.preventDefault();
+      return false;
+    }
+  }));
 
-//Shabad controller toggles
-$(".shabad-controller-toggle").on("click", shabadToggle);
+// Shabad controller toggles
+[...document.querySelectorAll('.shabad-controller-toggle')]
+  .forEach(e => e && e.addEventListener('click', shabadToggle));
 
-//Shabad display option toggles
-$(".display-option-toggle").on("click", displayOptionToggle);
+// Shabad display option toggles
+[...document.querySelectorAll('.display-option-toggle')]
+  .forEach(e => e && e.addEventListener('click', displayOptionToggle));
 
-$("#open-mobile-menu").on("click", function() {
-  document.body.classList.toggle("menu-open");
-})
-$(".top-bar-right .close a").on("click", function() {
-  document.body.classList.remove("menu-open");
-})
+[document.getElementById('open-mobile-menu')]
+  .forEach(e => e && addEventListener('click', () => document.body.classList.toggle('menu-open')));
 
-$("#open_share_menu").on("click", function() {
-  document.body.classList.toggle("share-open");
-})
+[document.querySelector('.top-bar-right .close a')]
+  .forEach(e => e && addEventListener('click', () => document.body.classList.remove('menu-open')));
 
-$("#search-options select").on("change", function() {
-  var update = $(this).data("update");
-  $("#" + update).val($(this).val());
-  if ($(this).attr("id") == "search-source") {
-    $("#top-bar-search-form").submit();
-  }
-});
+[document.getElementById('open_share_menu')]
+  .forEach(e => e && addEventListener('click', () => document.body.classList.toggle('share-open')));
 
-document.querySelector(".gurmukhi-keyboard-toggle").addEventListener("click", function() {
-  document.querySelector('.gurmukhi-keyboard').classList.toggle('active');
-});
+[...document.querySelectorAll('#search-options select')]
+  .forEach(e => e && e.addEventListener('change', function () {
+    const update = this.dataset.update;
+    document.getElementById(update).value = this.value;
+    if (this.getAttribute('id') === 'search-source') {
+      document.getElementById('top-bar-search-form').submit();
+    }
+  }));
+
+[document.querySelector('.gurmukhi-keyboard-toggle')]
+  .forEach(e => e && e.addEventListener('click', () => document.querySelector('.gurmukhi-keyboard').classList.toggle('active')));
 
-$("#shabad").on("click", ".share .copy", function() {
-  let el = $(this).parents(".line").children("textarea");
+$('#shabad').on('click', '.share .copy', function() {
+  const el = $(this).parents('.line').children('textarea');
   el.show().select();
-  document.execCommand("copy");
+  document.execCommand('copy');
   el.blur().hide();
 });
-$("#shabad").on("click", ".share .twitter", function() {
-  let tweet = $(this).parents(".line").children("textarea").val();
-  let shortURL = '\n'+shortenURL();
+
+$('#shabad').on('click', '.share .twitter', function () {
```

Comment:
Only this part is left in jQuery, I believe.
I'm not sure what the 2nd arg to `on` does, and how it translates to `querySelector`.

```js
  document
    .querySelector('#shabad')
    .addEventListener('click', e => {
      const $node = e.currentTarget;
      if ($node.parentNode.classList.contains('share') && $node.classList.contains('twitter')) {
        // do the thing?
      }
    });
```

Reply:
`querySelector` equivalent would be `’#shabad .share .twitter’`

Reply:
I think I'll take up this change in next sprint as the details within this click listener also makes use of some jQuery features that aren't that trivial to port. Can we go ahead with this for now?

Reply:
Sounds good. I will review again during my lunch break